pkg/http_endpoint/v1/handlers/http_grafana.go

```
@@ -0,0 +1,337 @@
+type V1UsersOrgs struct{}
+
+// LoginErrorCheck handles errors for login
+func LoginErrorCheck(err error) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		switch err.(type) {
+		// InvalidUsernamePassword means, that user provided invalid
+		// username/password. We have to return 401 error then
+		case common.InvalidUsernamePassword:
+			common.WriteErrorToResponse(w, http.StatusUnauthorized,
+				http.StatusText(http.StatusUnauthorized), err.Error())
+			return
+		// If any other error happened -> return 500 error
+		default:
+			log.Logger.Error(err)
+			common.WriteErrorToResponse(w, http.StatusInternalServerError,
+				http.StatusText(http.StatusInternalServerError),
+				"Internal server error occured")
+			return
+		}
+	}
+}
+
+// GetUsers get the list of users
+func GetUsers(clients *common.ClientContainer, handler common.HandlerInterface) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		users, err := handler.GetUsers(clients, w)
+		if err != nil {
+			log.Logger.Error(err)
+			common.WriteErrorToResponse(w, http.StatusInternalServerError,
+				http.StatusText(http.StatusInternalServerError),
+				"Internal server error occured")
+			return
+		}
+		userlist, _ := json.Marshal(users)
+		w.WriteHeader(http.StatusOK)
+		w.Write(userlist)
+	}
+}
+
+// GetUsersID gets the user details by ID
+func GetUsersID(clients *common.ClientContainer, handler common.HandlerInterface) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		userID := chi.URLParam(r, "userID")
+		ID, err := strconv.Atoi(userID)
+		if err != nil {
+			log.Logger.Error(err)
+			common.WriteErrorToResponse(w, http.StatusInternalServerError,
+				http.StatusText(http.StatusInternalServerError),
+				"ID provided is not integer")
+			return
+		}
+		userlist, err := handler.GetUserID(clients, w, ID)
+		if err != nil {
+			log.Logger.Error(err)
+			return
+		}
+		users, _ := json.Marshal(userlist)
```

serialization error is not handled here